src/forms/stock-dispense/useDispenseStock.tsx

```diff
@@ -0,0 +1,78 @@
+import useSWR from 'swr';
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { type InventoryItem, type MedicationDispense } from '../../types';
+import { getUuidFromReference } from '../../utils';
+
+//TODO: Add configuration to retrieve the stock dispense endpoint
+/**
+ * Fetches the inventory items for a given drug UUID.
+ *
+ * @param {string} drugUuid - The UUID of the drug.
+ * @returns {Array} - The inventory items.
+ */
+export const useDispenseStock = (drugUuid: string) => {
+  const url = `/ws/rest/v1/stockmanagement/stockiteminventory?v=default&limit=10&totalCount=true&drugUuid=${drugUuid}&includeBatchNo=true`;
+  const { data, error, isLoading } = useSWR<{ data: { results: Array<InventoryItem> } }>(url, openmrsFetch);
+  return { inventoryItems: data?.data?.results ?? [], error, isLoading };
+};
+
+/**
+ * Interface for the stock dispense request object.
+ */
+export type StockDispenseRequest = {
+  locationUuid: string;
+  patientUuid: string;
```

Comment:
this is just patient,order,encounter,stockItem,stockBatch as per new version of stockmanagement 1.4.0